hw/ip/otbn/doc/checklist.md

```diff
@@ -75,10 +75,12 @@ Security      | [SEC_RND_CNST][]        | Done        |
 
  Type         | Item                         | Resolution  | Note/Collaterals
 --------------|------------------------------|-------------|------------------
-Security      | [SEC_CM_ASSETS_LISTED][]     | Not Started |
-Security      | [SEC_CM_IMPLEMENTED][]       | Not Started |
-Security      | [SEC_CM_RTL_REVIEWED][]      | Not Started |
-Security      | [SEC_CM_COUNCIL_REVIEWED][]  | Not Started |
+Security      | [SEC_CM_ASSETS_LISTED][]     | Done        |
+Security      | [SEC_CM_IMPLEMENTED][]       | Done        |
+Security      | [SEC_CM_NON_RESET_FLOPS][]   | Done        | Reset flops are used for security critical flops (register file and ISPRs) as they need to be reset to something with valid integrity bits. OTBN performs a secure wipe at the end of any operation providing a limited window for an attacker to trigger a reset to produce leakage of sensitive data.
```

Comment:
The DV code shouldn't ever make that mistake: we've got a notion of "places that have an architectural value" that should handle this in the same way as it handles partially initialised memories.

Reply:
I think that according to [our non-reset vs. reset flops rationale](https://github.com/lowRISC/opentitan/issues/2603) the OTBN register files are a good example for the use of non-reset flops. The main problem is not the leakage occurring while resetting the register. We are more worried about what's happening afterwards. For example if a register holds a random mask to be applied to some other register (remember that the masking is implemented in SW here) and an attacker manages to reset that register while OTBN is running, the masking is effectively switched off. I believe the integrity bits don't really help here because the reset line is also connected to these. Synthesis will just insert set instead of reset flops for these bits. I am proposing to switch to non-reset flops for the register files and the ISPRs + RND/URND paths (incl. RND cache). WDYT @cdgori , @tjaychen, @zi-v? 

I just ran a regression with the reset removed from the register files (ISPRs seems to be more complicated). About half of the tests are failing with that. So it's probably not that easy. I can look into this if we agree to remove the reset from these registers.

Reply:
Hm your failures could be due to the ECC scheme. I.e., if the registers are not explicitly reset (either by SW or HW), this may create failures further down the road (you may recall that we ran into such complications when adjusting the ECC scheme last December).

Reply:
Yes, that's right @msfschaffner . Now, I remember having created an issues about that here #9751. @andreaskurth has now been looking into fixing DV accordingly and things look better now. There is still one issue in the Verilator smoke test but I think also this can be fixed. For details: #13192

Reply:
@vogelpi this makes sense to me. I guess the main argument is that these things aren't holding secrets per se, but removing resets allows us to remove a path where the entropy could be silenced.  I guess this somewhat argues for other entropy paths in the system being non-reset as well. What do you think?

Do you want to put `otbn` regfile as an example in the doc? (sorry I don't remember if we've already done that).

Reply:
yeah i was also thinking about the `prim_edn` interfaces.